android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java

```diff
@@ -687,9 +687,11 @@ public double getLastUpdateTimeSync() {
   @ReactMethod(isBlockingSynchronousMethod = true)
   public String getDeviceNameSync() {
     try {
-      String bluetoothName = Settings.Secure.getString(getReactApplicationContext().getContentResolver(), "bluetooth_name");
-      if (bluetoothName != null) {
-        return bluetoothName;
+      if (Build.VERSION.SDK_INT <= 31) {
+        String bluetoothName = Settings.Secure.getString(getReactApplicationContext().getContentResolver(), "bluetooth_name");
+        if (bluetoothName != null) {
+          return bluetoothName;
+        }
```

Comment:
should return value be set here if the condition `bluetoothName != null`  is not met?

Reply:
It's handled a few lines lower outside of the try/catch block. If neither of these blocks short circuit and return early, the last thing that happens is it returns 'unknown'.

Reply:
ah, I see